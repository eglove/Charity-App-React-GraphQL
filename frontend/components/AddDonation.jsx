import React, {Component} from "react";
import {Mutation} from "react-apollo";
import gql from "graphql-tag";
import styled from "styled-components";
import Error from "./ErrorMessage";
import Form from "./styles/Form";
import {CURRENT_USER_QUERY} from "./User";

const ADD_DONATION_MUTATION = gql`
    mutation ADD_DONATION_MUTATION (
        $id: ID!,
        $amount: Float!,
        $yearDonated: Int!,
    ) {
        addDonation(
            id: $id,
            amount: $amount,
            yearDonated: $yearDonated,
        ) {
            id
            amount
        }
    }
`;

const Required = styled.span`
    color: red;
`;

class AddDonation extends Component {
    state = {
        id: this.props.id,
        amount: 0,
        yearDonated: '',
    };

    handleChange = e => {
        const {name, type, value} = e.target;
        const val = type === 'number' ? parseFloat(value) : value;
        this.setState({[name]: val});
    };

    render() {
        return (
            <Mutation
                mutation={ADD_DONATION_MUTATION}
                variables={this.state}
                refetchQueries={[{query: CURRENT_USER_QUERY}]}
            >
                {(addDonation, {loading, error}) => (
                    <details>
                        <summary>Add Donation</summary>
                        <Form onSubmit={async (e) => {
                            e.preventDefault();
                            const res = await addDonation();
                        }}>
                            <Error error={error}/>
                            <fieldset disabled={loading} aria-busy={loading}>
                                <label htmlFor="amount">
                                    Amount <Required>*</Required>
                                    <input
                                        type="number"
                                        step="0.01"
                                        id="amount"
                                        name="amount"
                                        placeholder="Amount"
                                        onChange={this.handleChange}
                                        required
                                    />
                                </label>
                                <label htmlFor="yearDonated">
                                    Amount <Required>*</Required>
                                    <input
                                        type="number"
                                        id="yearDonated"
                                        name="yearDonated"
                                        placeholder="Year Donated"
                                        onChange={this.handleChange}
                                        required
                                    />
                                </label>
                                <button type="submit">Add{loading ? 'ing' : ''} Donation</button>
                            </fieldset>
                        </Form>
                    </details>
                )}
            </Mutation>
        )
    }
}

export default AddDonation;
export {ADD_DONATION_MUTATION};