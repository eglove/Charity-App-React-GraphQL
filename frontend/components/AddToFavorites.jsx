import React from "react";
import {Mutation} from "react-apollo";
import gql from "graphql-tag";
import {CURRENT_USER_QUERY} from "./User";

const ADD_TO_FAVORITES_MUTATION = gql`
    mutation addToFavorites($id: ID!) {
        addToFavorites(id: $id) {
            id
        }
    }
`;

class AddToFavorites extends React.Component {
    render() {
        const {id} = this.props;

        return (
            <Mutation
                mutation={ADD_TO_FAVORITES_MUTATION}
                variables={{id}}
                refetchQueries={[{query: CURRENT_USER_QUERY}]}
            >
                {(addToFavorites, {loading}) => (
                    <button disabled={loading} aria-busy={loading} onClick={addToFavorites}>ðŸ’™
                        Add{loading ? 'ing' : ''} To Favorites ðŸ’™</button>
                )}
            </Mutation>
        );
    }
}

export default AddToFavorites;
