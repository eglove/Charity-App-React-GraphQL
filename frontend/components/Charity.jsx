import Link from 'next/link';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import FavoritesButtonList from "./FavoritesButtonList";
import ItemStyles from "./styles/ItemStyles";

class Charity extends Component {

    static propTypes = {
        charity: PropTypes.object.isRequired,
    };

    render() {
        const {charity} = this.props;
        return (
            <ItemStyles>
                <Link href={{
                    pathname: '/charity',
                    query: {id: charity.id}
                }}>
                    <a>{charity.image && <img src={charity.image} alt={charity.imageDescription}/>}</a>
                </Link>
                <p>{charity.description.substring(0, 150)}...</p>
                <FavoritesButtonList charity={charity}/>
            </ItemStyles>
        );
    }
}

export default Charity;