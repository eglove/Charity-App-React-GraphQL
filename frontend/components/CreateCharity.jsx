import React, {Component} from 'react';
import {Mutation} from 'react-apollo';
import gql from "graphql-tag";
import styled from "styled-components";
import Router from "next/router";
import Head from 'next/head';
import Error from './ErrorMessage';
import Form from "./styles/Form";

const CREATE_CHARITY_MUTATION = gql`
    mutation CREATE_CHARITY_MUTATION(
        $ein: String!
        $name: String!
        $description: String!
        $website: String
        $image: String
        $largeImage: String
        $imageDescription: String
        $street: String
        $city: String
        $state: String
        $zip: String
    ) {
        createCharity(
            ein: $ein
            name: $name
            description: $description
            website: $website
            image: $image
            largeImage: $largeImage
            imageDescription: $imageDescription
            street: $street
            city: $city
            state: $state
            zip: $zip
        ) {
            id
        }
    }
`;

const Required = styled.span`
    color: red;
`;
const LookupLink = styled.a`
    text-decoration: underline;
`;

class CreateCharity extends Component {
    state = {
        ein: '',
        name: '',
        description: '',
        website: '',
        image: '',
        largeImage: '',
        imageDescription: '',
        street: '',
        city: '',
        state: '',
        zip: ''
    };

    handleChange = (e) => {
        const {name, type, value} = e.target;
        const val = type === 'number' ? parseFloat(value) : value;
        this.setState({[name]: val});
    };

    uploadFile = async (e) => {
        const files = e.target.files;
        const data = new FormData();
        data.append('file', files[0]);
        // Preset to resize images via cloudinary
        data.append('upload_preset', 'CharityApp');

        const res = await fetch(
            'https://api.cloudinary.com/v1_1/eglove/image/upload',
            {
                method: 'POST',
                body: data
            }
        );
        const file = await res.json();
        this.setState({
            image: file.secure_url,
            largeImage: file.eager[0].secure_url,
        })
    };

    render() {
        return (
            <Mutation mutation={CREATE_CHARITY_MUTATION} variables={this.state}>
                {(createCharity, {loading, error}) => (
                    <>
                    <Head>
                        <title>Cognitame ðŸ’™ Add Charity</title>
                    </Head>
                    <Form
                        onSubmit={async (e) => {
                            // Stop form from submitting
                            e.preventDefault();
                            // call mutation
                            const res = await createCharity();
                            // redirect to charity page
                            await Router.push({
                                pathname: '/charity',
                                query: {id: res.data.createCharity.id}
                            })
                        }}
                    >
                        <Error error={error}/>
                        <fieldset disabled={loading} aria-busy={loading}>

                            <label htmlFor="ein">
                                EIN (<LookupLink href="https://apps.irs.gov/app/eos/">Employer Identification
                                Number</LookupLink>) <Required>*</Required>
                                <input
                                    type="text"
                                    id="ein"
                                    name="ein"
                                    placeholder="EIN"
                                    value={this.state.ein}
                                    onChange={this.handleChange}
                                    required
                                />
                            </label>
                            <label htmlFor="name">
                                Name <Required>*</Required>
                                <input
                                    type="text"
                                    id="name"
                                    name="name"
                                    placeholder="Name"
                                    value={this.state.name}
                                    onChange={this.handleChange}
                                    required
                                />
                            </label>
                            <label htmlFor="description">
                                Description <Required>*</Required>
                                <textarea
                                    id="description"
                                    name="description"
                                    placeholder="Description"
                                    value={this.state.description}
                                    onChange={this.handleChange}
                                    required
                                />
                            </label>
                            <label htmlFor="website">
                                Website
                                <input
                                    type="text"
                                    id="website"
                                    name="website"
                                    placeholder="Website"
                                    value={this.state.website}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="file">
                                Image
                                <input
                                    type="file"
                                    id="file"
                                    name="file"
                                    placeholder="Upload Logo"
                                    onChange={this.uploadFile}
                                />
                                {this.state.image && (<img src={this.state.image} width="200" alt="Image Upload Preview"/>)}
                            </label>
                            <label htmlFor="imageDescription">
                                Image Description
                                <textarea
                                    id="imageDescription"
                                    name="imageDescription"
                                    placeholder="Image Description"
                                    value={this.state.imageDescription}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="street">
                                Street
                                <input
                                    type="text"
                                    id="street"
                                    name="street"
                                    placeholder="Street"
                                    value={this.state.street}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="city">
                                City
                                <input
                                    type="text"
                                    id="city"
                                    name="city"
                                    placeholder="City"
                                    value={this.state.city}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="state">
                                State
                                <input
                                    type="text"
                                    id="state"
                                    name="state"
                                    placeholder="State"
                                    value={this.state.state}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <label htmlFor="zip">
                                Zip
                                <input
                                    type="text"
                                    id="zip"
                                    name="zip"
                                    placeholder="Zip"
                                    value={this.state.zip}
                                    onChange={this.handleChange}
                                />
                            </label>
                            <button type="submit">Submit</button>
                        </fieldset>
                    </Form>
                    </>
                )}
            </Mutation>
        );
    }
}

export default CreateCharity;
export {CREATE_CHARITY_MUTATION};