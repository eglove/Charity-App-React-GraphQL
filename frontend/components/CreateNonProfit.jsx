import React, {Component} from 'react';
import gql from 'graphql-tag'
import {Mutation} from 'react-apollo';
import Router from 'next/router';
import Error from './ErrorMessage';

const CREATE_NONPROFIT_MUTATION = gql`
    mutation CREATE_NONPROFIT_MUTATION(
        $ein: Int!
        $name: String!
        $ico: String
        $street: String!
        $city: String!
        $state: String!
        $zip: String!
        $group: Int
        $subsection: Int
        $affiliation: Int
        $classification: Int
        $ruling: Int
        $deductibility: Int
        $foundation: Int
        $activity: Int
        $organization: Int
        $status: Int
        $taxperiod: Int
        $assetCd: Int
        $incomeCd: Int
        $filingReqCd: Int
        $pfFilingReqCd: Int
        $acctPd: Int
        $assetAmt: Int
        $revenueAmt: Int
        $nteeCd: String
        $sortName: String
    ) {
        createNonProfit(data: {
            ein: $ein
            name: $name
            ico: $ico
            street: $street
            city: $city
            state: $state
            zip: $zip
            group: $group
            subsection: $subsection
            affiliation: $affiliation
            classification: $classification
            ruling: $ruling
            deductibility: $deductibility
            foundation: $foundation
            activity: $activity
            organization: $organization
            status: $status
            taxperiod: $taxperiod
            assetCd: $assetCd
            incomeCd: $incomeCd
            filingReqCd: $filingReqCd
            pfFilingReqCd: $pfFilingReqCd
            acctPd: $acctPd
            assetAmt: $assetAmt
            revenueAmt: $revenueAmt
            nteeCd: $nteeCd
            sortName: $sortName
        }) {
            id
        }
    }
`;

class CreateNonProfit extends Component {
    state = {
        ein: 0,
        name: '',
        street: '',
        city: '',
        state: '',
        zip: '',
        ico: '',
        group: 0,
        subsection: 0,
        affiliation: 0,
        classification: 0,
        ruling: 0,
        deductibility: 0,
        foundation: 0,
        activity: 0,
        organization: 0,
        status: 0,
        taxperiod: 0,
        assetCd: 0,
        incomeCd: 0,
        filingReqCd: 0,
        pfFilingReqCd: 0,
        acctPd: 0,
        assetAmt: 0,
        revenueAmt: 0,
        nteeCd: '',
        sortName: ''
    };

    handleChange = (e) => {
        const {name, type, value} = e.target;
        const val = type === 'number' ? parseFloat(value) : value;
        this.setState({[name]: val})
    };

    render() {
        return (
            <div className="component">
                <Mutation mutation={CREATE_NONPROFIT_MUTATION} variables={this.state}>
                    {(createNonProfit, {loading, error}) => (
                        <form onSubmit={async e => {
                            e.preventDefault();
                            const res = await createNonProfit();
                            console.log(res);
                            await Router.push(`/nonProfit?id=${res.data.createNonProfit.id}`);
                        }}>
                            <Error error={error}/>
                            <fieldset disabled={loading} aria-busy={loading}>
                                <label htmlFor="ein">
                                    EIN:&emsp;<input
                                    type="number"
                                    id="ein"
                                    name="ein"
                                    placeholder="EIN"
                                    required
                                    value={this.state.ein}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <br/>
                                <label htmlFor="name">
                                    Name:&emsp;<input
                                    type="text"
                                    id="name"
                                    name="name"
                                    placeholder="Name"
                                    required
                                    value={this.state.name}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <br/>
                                <label htmlFor="street">
                                    Street:&emsp;<input
                                    type="text"
                                    id="street"
                                    name="street"
                                    placeholder="Street"
                                    required
                                    value={this.state.street}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <br/>
                                <label htmlFor="city">
                                    City:&emsp;<input
                                    type="text"
                                    id="city"
                                    name="city"
                                    placeholder="City"
                                    required
                                    value={this.state.city}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <br/>
                                <label htmlFor="state">
                                    State:&emsp;<input
                                    type="text"
                                    id="state"
                                    name="state"
                                    placeholder="State"
                                    required
                                    value={this.state.state}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <br/>
                                <label htmlFor="zip">
                                    Zip Code:&emsp;<input
                                    type="text"
                                    id="zip"
                                    name="zip"
                                    placeholder="Zip Code"
                                    required
                                    value={this.state.zip}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <p>Optional</p>
                                <label htmlFor="ico">
                                    ICO:&emsp;<input
                                    type="text"
                                    id="ico"
                                    name="ico"
                                    placeholder="ICO"
                                    value={this.state.ico}
                                    onChange={this.handleChange}
                                />
                                </label>
                                <br/>
                                <button type="submit">Add</button>
                            </fieldset>
                        </form>
                    )}
                </Mutation>
            </div>
        );
    }
}

export default CreateNonProfit;
export {CREATE_NONPROFIT_MUTATION};