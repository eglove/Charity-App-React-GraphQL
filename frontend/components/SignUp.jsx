import React, {Component} from 'react';
import {Mutation} from "react-apollo";
import gql from "graphql-tag";
import Head from 'next/head';
import Form from "./styles/Form";
import Error from './ErrorMessage';
import {CURRENT_USER_QUERY} from "./User";

const SIGNUP_MUTATION = gql`
    mutation SIGNUP_MUTATION(
        $name: String!
        $email: String!
        $password: String!
    ) {
        signup (
            name: $name,
            email: $email,
            password: $password
        ) {
            id
            email
            name
        }
    }
`;

class SignUp extends Component {

    state = {
        name: '',
        email: '',
        password: '',
    };

    saveToState = (e) => {
        this.setState({[e.target.name]: e.target.value});
    };

    render() {
        return (
            <Mutation
                mutation={SIGNUP_MUTATION}
                variables={this.state}
                refetchQueries={[{CURRENT_USER_QUERY}]}
            >
                {(signup, {error, loading}) => {
                    return (
                        <>
                            <Head>
                                <title>Cognitame üëè SignUp</title>
                            </Head>
                        <Form
                            method="POST"
                            onSubmit={async e => {
                                e.preventDefault();
                                await signup();
                                this.setState({name: '', email: '', password: ''});
                            }}
                        >
                            <fieldset disabled={loading} aria-busy={loading}>
                                <h2>Sign Up For An Account</h2>
                                <Error error={error}/>
                                <label htmlFor="name">
                                    Name
                                    <input
                                        type="text"
                                        name="name"
                                        placeholder="Name"
                                        value={this.state.name}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <label htmlFor="email">
                                    Email
                                    <input
                                        type="email"
                                        name="email"
                                        placeholder="Email"
                                        value={this.state.email}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <label htmlFor="password">
                                    Password
                                    <input
                                        type="password"
                                        name="password"
                                        placeholder="Password"
                                        value={this.state.password}
                                        onChange={this.saveToState}
                                    />
                                </label>
                                <button type="submit">Sign{loading ? 'ing' : ''} Up!</button>
                            </fieldset>
                        </Form>
                        </>
                    )
                }}
            </Mutation>
        );
    }
}

export default SignUp;