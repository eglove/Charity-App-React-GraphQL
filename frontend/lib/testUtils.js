import casual from 'casual';

// seed it so we get consistent results
casual.seed(777);

const fakeCharity = () => ({
  __typename: 'Charity',
  id: casual.uuid,
  ein: casual.uuid,
  name: casual.company_name,
  description: casual.description,
  website: casual.url,
  image: `${casual.word}.png`,
  largeImage: `${casual.word}.png`,
  imageDescription: casual.description,
  street: casual.street,
  city: casual.city,
  state: casual.state,
  zip: casual.zip,
  user: casual.uuid,
});

const fakeUser = () => ({
  __typename: 'User',
  id: casual.uuid,
  name: casual.name,
  email: casual.email,
  password: casual.password,
  totalDonated: casual.random,
  permissions: ['ADMIN'],
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export {
  LocalStorageMock,
  fakeCharity,
  fakeUser,
};
