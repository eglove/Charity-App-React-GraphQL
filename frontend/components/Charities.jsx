import React, {Component} from 'react';
import {Query} from "react-apollo";
import gql from "graphql-tag";
import styled from "styled-components";
import Charity from "./Charity";
import Pagination from "./Pagination";
import {perPage} from "../config";

const ALL_CHARITIES_QUERY = gql`
    query ALL_CHARITIES_QUERY($skip: Int = 0, $first: Int = ${perPage}) {
        charities(first: $first, skip: $skip, orderBy: name_ASC) {
            id
            ein
            name
            description
            website
            image
            largeImage
            imageDescription
            street
            city
            state
            zip
        }
    }
`;

const Center = styled.div`
    text-align: center;
`;

const CharitiesList = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 60px;
    max-width: ${props => props.theme.maxWidth};
    margin: 0 auto;
`;

class Charities extends Component {
    render() {
        return (
            <Center>
                <Pagination page={this.props.page}/>
                <Query
                    query={ALL_CHARITIES_QUERY}
                    // No good solution to keep items list up to date with server
                    // without sacrificing performance
                    // fetchPolicy="network-only"
                    variables={{
                        skip: this.props.page * perPage - perPage,
                    }}
                >
                    {({data, error, loading}) => {
                        if(loading) return <p>Loading...</p>;
                        if(error) return <p>Error: {error.message}</p>
                        return (
                            <CharitiesList>
                                {data.charities.map(charity =>
                                    <Charity key={charity.id} charity={charity}/>
                                )}
                            </CharitiesList>
                        )
                    }}
                </Query>
                <Pagination page={this.props.page}/>
            </Center>
        );
    }
}

export default Charities;
export {ALL_CHARITIES_QUERY};