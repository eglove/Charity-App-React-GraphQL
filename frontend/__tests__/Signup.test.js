import {mount} from "enzyme";
import SignUp, {SIGNUP_MUTATION} from "../components/SignUp";
import {CURRENT_USER_QUERY} from "../components/User";
import {fakeUser} from "../lib/testUtils";
import {MockedProvider} from "react-apollo/test-utils";
import toJson from "enzyme-to-json";
import {ApolloConsumer} from "react-apollo";
import wait from "waait";

function type(wrapper, name, value) {
    wrapper.find(`input[name="${name}"]`).simulate('change', {
        target: {name, value},
    });
}

const me = fakeUser();
// signup mock mutation
const mocks = [
    {
        request: {
            query: SIGNUP_MUTATION,
            variables: {
                email: me.email,
                name: me.name,
                password: 'fakePassword',
                totalDonated: me.totalDonated,
            },
        },

        result: {
            data: {
                signup: {
                    __typename: 'User',
                    id: 'abc123',
                    email: me.email,
                    name: me.name,
                    password: 'fakePassword',
                    totalDonated: me.password,
                },
            },
        },
    },

    // current user query mock
    {
        request: {query: CURRENT_USER_QUERY},
        result: {data: {me}},
    },
];

describe('<Signup/>', () => {
    it('renders matches snapshot', async () => {
        const wrapper = mount(
            <MockedProvider>
                <SignUp/>
            </MockedProvider>
        );
        expect(toJson(wrapper.find('form'))).toMatchSnapshot();
    });

    xit('calls the mutation properly', async () => {
        let apolloClient;
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <ApolloConsumer>
                    {client => {
                        apolloClient = client;
                        return <SignUp/>
                    }}
                </ApolloConsumer>
            </MockedProvider>
        );

        await wait();
        wrapper.update();
        type(wrapper, 'email', me.email);
        type(wrapper, 'name', me.name);
        type(wrapper, 'password', 'fakePassword');
        type(wrapper, 'totalDonated', me.totalDonated);

        wrapper.find('form').simulate('submit');
        await wait();
        // query the user of the apollo client
        const user = await apolloClient.query({query: CURRENT_USER_QUERY});
        expect(user.data.me).toMatchObject(me);
    });
});
