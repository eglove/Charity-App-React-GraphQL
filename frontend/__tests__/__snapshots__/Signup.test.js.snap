// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> renders matches snapshot 1`] = `
<form
  className="Form-feif0j-0 hNIufj"
  method="POST"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Sign Up For An Account
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="name"
    >
      Name
      <input
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Sign
       Up!
    </button>
  </fieldset>
  <Link
    href="signin"
    style="text-align:center"
  >
    <a
      href="signin"
      onClick={[Function]}
    >
      Sign into Existing Account
    </a>
  </Link>
  <br />
  <Link
    href="resetRequest"
    style="text-align:center"
  >
    <a
      href="resetRequest"
      onClick={[Function]}
    >
      Reset Password
    </a>
  </Link>
</form>
`;
